File: src/v/iceberg/transform.cc
// Copyright 2024 Redpanda Data, Inc.
//
// Use of this software is governed by the Business Source License
// included in the file licenses/BSL.md
//
// As of the Change Date specified in that file, in accordance with
// the Business Source License, use of this software will be governed
// by the Apache License, Version 2.0
#include "iceberg/transform.h"

namespace iceberg {

struct transform_comparison_visitor {
    template<typename T, typename U>
    bool operator()(const T&, const U&) const {
        static_assert(!std::is_same<T, U>::value);
        return false;
    }
    bool
    operator()(const bucket_transform& lhs, const bucket_transform& rhs) const {
        return lhs.n == rhs.n;
    }
    bool operator()(
      const truncate_transform& lhs, const truncate_transform& rhs) const {
        return lhs.length == rhs.length;
    }
    template<typename T>
    bool operator()(const T&, const T&) const {
        return true;
    }
};

bool operator==(const transform& lhs, const transform& rhs) {
    return std::visit(transform_comparison_visitor{}, lhs, rhs);
}

} // namespace iceberg

File: src/go/rpk/pkg/cli/transform/transform.go
// Copyright 2023 Redpanda Data, Inc.
//
// Use of this software is governed by the Business Source License
// included in the file licenses/BSL.md
//
// As of the Change Date specified in that file, in accordance with
// the Business Source License, use of this software will be governed
// by the Apache License, Version 2.0

package transform

import (
	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
	"github.com/spf13/afero"
	"github.com/spf13/cobra"
)

func NewCommand(fs afero.Fs, p *config.Params, execFn func(string, []string) error) *cobra.Command {
	cmd := &cobra.Command{
		Use:     "transform",
		Aliases: []string{"wasm", "transfrom"}, //nolint:misspell // auto correct a common misspelling
		Short:   "Develop, deploy and manage Redpanda data transforms",
	}
	p.InstallKafkaFlags(cmd)
	cmd.AddCommand(
		newDeployCommand(fs, p),
		newDeleteCommand(fs, p),
		newListCommand(fs, p),
		newInitializeCommand(fs),
		newBuildCommand(fs, execFn),
		newLogsCommand(fs, p),
		newPauseCommand(fs, p),
		newResumeCommand(fs, p),
	)
	return cmd
}
